packages/headless/src/composables/slider/createSlider.ts

```diff
@@ -0,0 +1,861 @@
+
+/**
+ * Composable for creating an accessibility-compliant slider.
+ * For supported keyboard shortcuts, see: https://www.w3.org/WAI/ARIA/apg/patterns/slider/
+ */
+export const createSlider = createBuilder((options: CreateSliderOptions) => {
+  const sliderRef = createElRef<HTMLElement>();
+
+  // Computed properties from options
+  const min = computed(() => unref(options.min) ?? 0);
+  const max = computed(() => unref(options.max) ?? 100);
+  const values = computed(() => unref(options.values) ?? [min.value ?? 0]);
+  const step = computed(() =>
+    typeof unref(options.step) === "undefined" ? 1 : unref(options.step),
+  );
+  const shiftStep = computed(() => unref(options.shiftStep) ?? 10);
+  const isDisabled = computed(() => unref(options.disabled) ?? false);
+  const marks = computed(() => unref(options.marks) ?? false);
+  const label = computed(() => unref(options.label));
+  const orientation = computed(() => unref(options.orientation) ?? "horizontal");
+
+  // Refs and variables for internal state
+  /**
+   * Holds the identifier of the active touch point (Touch.identifier).
+   * Reason: lets the slider track and finish the drag with the *same finger*
+   * only—avoiding interference from other touches in multi-touch scenarios and
+   * preventing stuck or incorrect drags.
+   */
+  let touchId: number | null = null;
+  /**
+   * Counts the number of move events during a drag operation.
+   * Reason: lets us differentiate between intentional drags and accidental slight movements,
+   */
+  let movesSinceStart: number = 0;
+  /**
+   * Used for committing the value when the user stops dragging accidentally.
+   * Sometimes the committed value is different from the current value while dragging.
+   * So we need to keep track of it.
+   */
+  let lastChangedValue: number[] | null = null;
+  /**
+   * Previous active thumb index during drag. Used during drag to keep the same thumb active.
+   */
+  let previousActiveIndex: number | null = null;
+  /**
+   * Indicates whether the slider is currently being dragged or not.
+   */
+  const isDragging = ref<boolean>(false);
+  /**
+   * Index of the currently active thumb.
+   * Thumb could be active even if not dragging (e.g. click on the rail to move the thumb there).
+   */
+  const activeThumbIndex = ref<number>(-1);
+  /**
+   * Index of the thumb that is currently focused.
+   */
+  const focusedThumbIndex = ref<number>(-1);
+  /**
+   * Indicates whether the slider is a range slider (with two or more thumbs) or a single value slider (with one thumb).
+   */
+  const isRange = computed(() => values.value.length > 1);
+
+  /**
+   * If `showMarks` is `true` and `step` is provided, marks will be generated for each step in the range.
+   * If `marks` is an array and `showMarks` is `true`, it will be used as provided.
+   * If `showMarks` is `false`, no marks will be shown.
+   */
+  const marksList = computed<{ value: number; label?: string }[]>(() => {
```

Comment:
```suggestion
  const marksList = computed<Exclude<SliderMark, number>[]>(() => {
```

So we don't need to maintain the type in two places :) Alternatively, we can also remove the `| number` from the type itself and define the marks option as `(SliderMark | number)[]`